### PR TITLE
Fix constand_keyword test run and properly test recent behavior change

### PR DESCRIPTION
**Pull request by lkts**
This is a follow-up to https://github.com/elastic/elasticsearch/pull/117182.

That PR didn't change yaml tests since they were not running and didn't introduce a test feature. Tests didn't run because legacy yaml rest tests don't provide tests features and the test was using previously introduced `mapper.source.mode_from_index_setting`. As a result this test was always skipped.

This PR introduces new test feature, fixes the test and migrates this project to proper yaml test runner.

### CODE REVIEW

File: x-pack/plugin/mapper-constant-keyword/src/main/module-info.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+module org.elasticsearch.xpack.constantkeyword {
```
Comment (lkts) [inline x-pack/plugin/mapper-constant-keyword/src/main/module-info.java:10]: I am not confident in this files contents.

File: x-pack/plugin/mapper-constant-keyword/build.gradle [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import org.elasticsearch.gradle.internal.info.BuildParams
 
 apply plugin: 'elasticsearch.internal-es-plugin'
-apply plugin: 'elasticsearch.legacy-yaml-rest-test'
+apply plugin: 'elasticsearch.internal-yaml-rest-test'
+apply plugin: 'elasticsearch.internal-test-artifact'
```
Comment (mark-vieira) [inline x-pack/plugin/mapper-constant-keyword/build.gradle:5]: The test artifact plugin shouldn't be required here. This is only necessarily if another project needs to consume the tests of this project as an artifact dependency.

File: x-pack/plugin/mapper-constant-keyword/src/main/module-info.java (hunk 1 above)
Comment (mark-vieira) [reply x-pack/plugin/mapper-constant-keyword/src/main/module-info.java:10]: @rjernst is modularizing this plugin strictly necessary or is the services info file sufficient?

**Review by mark-vieira — APPROVED** [review]
I'd confirm with @elastic/es-core-infra on the module stuff, but otherwise LGTM.

File: x-pack/plugin/mapper-constant-keyword/src/main/java/org/elasticsearch/xpack/constantkeyword/ConstantKeywordFeatures.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+package org.elasticsearch.xpack.constantkeyword;
+
+import org.elasticsearch.features.FeatureSpecification;
+import org.elasticsearch.features.NodeFeature;
+
+import java.util.Set;
+
+public class ConstantKeywordFeatures implements FeatureSpecification {
+    public static final NodeFeature SYNTHETIC_SOURCE_WRITE_FIX = new NodeFeature("constant_keyword.synthetic_source_write_fix");
```
Comment (dnhatn) [inline x-pack/plugin/mapper-constant-keyword/src/main/java/org/elasticsearch/xpack/constantkeyword/ConstantKeywordFeatures.java:16]: Can we add this to MapperFeatures instead? I think that should be okay, although the mapping type is in x-pack.

Comment (lkts) [reply x-pack/plugin/mapper-constant-keyword/src/main/java/org/elasticsearch/xpack/constantkeyword/ConstantKeywordFeatures.java:16]: We can i just thought this is cleaner.

Comment (dnhatn) [reply x-pack/plugin/mapper-constant-keyword/src/main/java/org/elasticsearch/xpack/constantkeyword/ConstantKeywordFeatures.java:16]: I think we can define a NodeFeature in MapperFeatures class?

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks Sasha!

**Comment by ldematte** [comment]
> the services info file sufficient?

It depends :) it would have been necessary (and sufficient) for tests that run un-modularized. Not 100% about rest tests; they run with a full cluster, so modular I think.
But I see it is gone :)

**Pull request merged**